types/module/module.go

```diff
@@ -42,10 +42,9 @@ import (
 	"github.com/spf13/cobra"
 	"golang.org/x/exp/maps"
 
-	storetypes "cosmossdk.io/store/types"
-
 	errorsmod "cosmossdk.io/errors"
-
+	storetypes "cosmossdk.io/store/types"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
```

Comment:
ideally we dont do this, this seems like a bandaid that we will have to clean up later on. i think we should identify how to fix this problem without allowing modules to override consensus params.

Reply:
We must be doing this within `BeginBlock` of `Manager` since the consensus params is accessed by other modules right after migration within the same block execution.

Reply:
Yeah I guess there's no way around it it seems :/

Reply:
We could perhaps return an `any` instead and then type cast where needed?

Reply:
not sure if get params by [consensusKeeper](https://github.com/cosmos/cosmos-sdk/pull/16583/commits/591d7cf9076778d34c11f0cfc42366dcb22a0283) is better, but we need add module one by one (not to mention module outside sdk)

Reply:
@tac0turtle May I ask if you have any better idea beside these two approaches?

Reply:
I think the core issue here is the migration is "system" level functionalities, probably shouldn't happen in a module in the first place, is it possible to move the migration logic into baseapp somehow?

Reply:
sorry for delay, the dependency is fine. agree with @yihuang. Maybe for this version of baseapp this is fine but for later on we can change it. Im not sure we have a clean alternative

Reply:
100% agree 👍